view/src/components/surplus.js

```diff
@@ -0,0 +1,555 @@
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  /**
+   * Given the name of a state updates the state to a new value provided
+   * as a a parameter
+   * @param name  Name of the state that will be change
+   * @param value New value for the state
+   */
+  handleAsyncChange = (name, value) => {
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  /** Returns the authentication token stored in local storage */
+  getAuth = () => {
+    authMiddleWare(this.props.history);
+    return localStorage.getItem("AuthToken");
+  };
+
+  /** Load in all of the current food banks when the component has mounted */
+  componentDidMount() {
+    axios.defaults.headers.common = { Authorization: `${this.getAuth()}` };
+    axios
+      .get("/surplus")
+      .then((response) => {
+        this.setState({
+          // surplus state
+          surplusObjects: response.data,
+          // page state
+          uiLoading: false,
+        });
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  /**
+   * Takes a food bank object as an input and deletes the given food bank
+   * object from the database
+   * @param data A food bank object
+   */
+  handleDelete(data) {
+    axios.defaults.headers.common = { Authorization: `${this.getAuth()}` };
+    let surplusId = data.surplus.surplusId;
+    axios
+      .delete(`surplus/${surplusId}`)
+      .then(() => {
+        window.location.reload();
+        this.props.alert("success", "Surplus successfully deleted!");
+      })
+      .catch((err) => {
+        console.error(err);
+        this.props.alert(
+          "error",
+          "An error occurred when attempting to delete the Surplus!"
+        );
+      });
+  }
+
+  /**
+   * Takes a food bank object as an input and opens a dialog page to
+   * allow the user to update the attributes of the food bank object
+   * @param data A food bank object
+   */
+  handleEditClick(data) {
+    this.setState({
+      // surplus state
+      surplusId: data.surplus.surplusId,
+      produceId: data.surplus.produceId,
+      produceName: data.surplus.produceName,
+      originFarmId: data.surplus.originFarmId,
+      originFarmName: data.surplus.originFarmContactName,
+      originaFarmLocation: data.surplus.originFarmLocation,
+      originFarmContactName: data.surplus.originFarmContactName,
+      originFarmContactPhone: data.surplus.originFarmContactPhone,
+      available: data.surplus.available,
+      cost: data.surplus.cost,
+      totalQuantityAvailable: data.surplus.totalQuantityAvailable,
+      packagingType: data.surplus.packagingType,
+      // page state
+      buttonType: "Edit",
+      open: true,
+    });
+  }
+
+  /**
+   * Takes a food bank object as an input and opens a popup with all the
+   * information about the food bank (currently not being used -> will be
+   * updated to show augmented information)
+   * @param data A food bank object
+   */
+  handleViewOpen(data) {
```

Comment:
Perhaps specifying this to be about the "detailedView" instead of view could be helpful, since at first glance, I thought it had to do with the regular viewing mode of the cards. (Since we might still have extra info to display in detailedView during MVP -> Alpha)